elasticapm/contrib/serverless/aws.py

```diff
@@ -367,21 +377,23 @@ def get_url_dict(event):
     Reconstruct URL from API Gateway
     """
     headers = event.get("headers", {})
-    proto = headers.get("X-Forwarded-Proto", "https")
-    host = headers.get("Host", "")
-    path = event.get("path", "")
-    port = headers.get("X-Forwarded-Port")
+    protocol = headers.get("X-Forwarded-Proto", headers.get("x-forwarded-proto", "https"))
```

Comment:
you could wrap `headers` in a [`wsgiref.headers.Headers`](https://docs.python.org/3/library/wsgiref.html#wsgiref.headers.Headers) object, which would handle the case-insensitiveness aspect of HTTP headers. It's a somewhat dumb implementation, that loops over all headers on a `.get()`. If you know that the two variants you handle are the only two possible, it's probably not worth the hassle.

Reply:
Super good to know, but in this case there are only a couple of instances (which only have two case variations) so I probably won't bother.